cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -244,22 +244,8 @@ func (a *StaticAutoscaler) cleanUpIfRequired() {
 
 func (a *StaticAutoscaler) initializeClusterSnapshot(nodes []*apiv1.Node, scheduledPods []*apiv1.Pod) caerrors.AutoscalerError {
 	a.ClusterSnapshot.Clear()
-
-	knownNodes := make(map[string]bool)
-	for _, node := range nodes {
-		if err := a.ClusterSnapshot.AddNode(node); err != nil {
-			klog.Errorf("Failed to add node %s to cluster snapshot: %v", node.Name, err)
-			return caerrors.ToAutoscalerError(caerrors.InternalError, err)
-		}
-		knownNodes[node.Name] = true
-	}
-	for _, pod := range scheduledPods {
-		if knownNodes[pod.Spec.NodeName] {
-			if err := a.ClusterSnapshot.AddPod(pod, pod.Spec.NodeName); err != nil {
-				klog.Errorf("Failed to add pod %s scheduled to node %s to cluster snapshot: %v", pod.Name, pod.Spec.NodeName, err)
-				return caerrors.ToAutoscalerError(caerrors.InternalError, err)
-			}
-		}
+	if err := a.ClusterSnapshot.Initialize(nodes, scheduledPods); err != nil {
```

Comment:
1. Is Clear() call needed here? Based on the docstring for Initialize(), I'd expect Clear() to be a part of Initialize().
2. Does it make sense to wrap a single method in a helper function? Feels like at this point we should just inline it back into RunOnce.

Reply:
Good catch, removed the helper method altogether